plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java

```diff
@@ -687,10 +687,13 @@ public Stream<SourceFile> parse(Project subproject, Set<Path> alreadyParsed, Exe
                         .collect(toList());
 
                 // The compilation classpath doesn't include the transitive dependencies
-                // So we use the runtime classpath to get complete type information
+                // The runtime classpath doesn't include compile only dependencies, e.g.: lombok, servlet-api
+                // So we use both together to get comprehensive type information
                 List<Path> dependencyPathsNonFinal;
                 try {
-                    dependencyPathsNonFinal = sourceSet.getRuntimeClasspath().getFiles().stream()
+                    dependencyPathsNonFinal = Stream.concat(
+                                    sourceSet.getRuntimeClasspath().getFiles().stream(),
+                                    sourceSet.getCompileClasspath().getFiles().stream())
```

Comment:
I do wonder if having the runtime classpath is accurate for the parsers. For the `compileJava` task as an example, it only receives the `compileClasspath` configuration.

Reply:
What you are saying makes sense to me. I found this when I was trying to track down why we no longer had any type attribution for Lombok annotations.

As a first fix it seems less drastic to revert to using compile + runtime classpath, but as you say, the compiler should probably only need the compile classpath, so I think we should definitely look into this as a follow-up.

Reply:
That sounds fine to me then. :)

Reply:
Please also note that the runtime classpath includes entries for `build/classes/java/main`, `build/classes/kotlin/main`, etc. This is potentially important for cross-language type attribution. Something to be investigated.

Reply:
That should be handled on the `compileClasspath` as well for the plugins that support cross source set configurations or where users have configured it already. Otherwise, the types wouldn't be able to be resolved by that particular language compiler (javac, groovyc, kotlinc, etc). Most of them don't allow full cross compilation except for the groovyc compiler which is the only one that will compile both .java and .groovy files directly.

Reply:
Interesting. Then this approach with only the compiler classpath sounds very compelling.